operator/validator/controller.go

```diff
@@ -718,54 +718,58 @@ func (c *controller) startValidator(v *validator.Validator) (bool, error) {
 	if v.Share.BeaconMetadata.Index == 0 {
 		return false, errors.New("could not start validator: index not found")
 	}
-	if err := v.Start(c.logger); err != nil {
+	started, err := v.Start(c.logger)
+	if err != nil {
 		c.metrics.ValidatorError(v.Share.ValidatorPubKey)
 		return false, errors.Wrap(err, "could not start validator")
 	}
-	c.recentlyStartedValidators.Add(1)
+	if started {
+		c.recentlyStartedValidators.Add(1)
+	}
 	return true, nil
 }
 
 // UpdateValidatorMetaDataLoop updates metadata of validators in an interval
 func (c *controller) UpdateValidatorMetaDataLoop() {
-	var interval = c.beacon.GetBeaconNetwork().SlotDurationSec()
-	go c.metadataUpdateQueue.Start()
+	var interval = c.beacon.GetBeaconNetwork().SlotDurationSec() * 2
 
-	lastUpdated := make(map[string]time.Time)
-	for {
-		time.Sleep(interval)
-		start := time.Now()
+	// Prepare share filters.
+	filters := []registrystorage.SharesFilter{
+		registrystorage.ByNotLiquidated(),
+	}
 
-		// Prepare filters.
-		filters := []registrystorage.SharesFilter{
-			registrystorage.ByNotLiquidated(),
-		}
+	// Filter for validators who belong to our operator.
+	if !c.validatorOptions.Exporter {
+		filters = append(filters, registrystorage.ByOperatorID(c.GetOperatorData().ID))
+	}
 
-		// Filter for validators who belong to our operator.
-		if !c.validatorOptions.Exporter {
-			filters = append(filters, registrystorage.ByOperatorID(c.GetOperatorData().ID))
-		}
+	// Filter for validators which haven't been updated recently.
+	filters = append(filters, func(s *ssvtypes.SSVShare) bool {
+		last, ok := c.metadataLastUpdated.Get(string(s.ValidatorPubKey))
+		return !ok || time.Since(last) > c.metadataUpdateInterval
+	})
 
-		// Filter for validators which haven't been updated recently.
-		filters = append(filters, func(s *ssvtypes.SSVShare) bool {
-			last, ok := lastUpdated[string(s.ValidatorPubKey)]
-			return !ok || time.Since(last) > c.metadataUpdateInterval
-		})
+	for {
+		time.Sleep(interval)
+		start := time.Now()
 
 		// Get the shares to fetch metadata for.
 		shares := c.sharesStorage.List(nil, filters...)
 		var pks [][]byte
 		for _, share := range shares {
 			pks = append(pks, share.ValidatorPubKey)
-			lastUpdated[string(share.ValidatorPubKey)] = time.Now()
+			c.metadataLastUpdated.Set(string(share.ValidatorPubKey), time.Now())
```

Comment:
Look at the comment I put on `metadataLastUpdated` on the top, this wasn't thread safe before, no need to add hashmap and make it thread safe now. map seems to be enough since it only used in this loop.